docs/first_semantic_game/02.simple_chat_npcs.rst

```diff
@@ -1,163 +1,395 @@
+            string errMsg = $"Could not get reply from {ChatName}: {ex.Message}";
+            GD.PrintErr(errMsg);
         }
     }
 
-Testing the Minimal `ChatAI` Class
-----------------------------------
 
-To verify the `ChatAI` minimal class, setup an interaction between two NPCs and allow them to communicate via `ReceiveMsg(msg)` method. Connect a handler function to the `ReplyGenerated` signal in order to observe the generated response, thus demonstrating an interactive conversation powered by Semantic Kernel.
+Two Arguing NPCs
+----------------
+
+In this section, you will learn how to set up NPCs named "East Troll" and 
+"West Troll" to automatically argue with each other using Godot's properties
+panel. These trolls are configured to have a debate about which side of the 
+river is better.
+
+Creating the East and West Troll Nodes
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. **Create an NPC Scene**: Drag the ``NPC`` node from the main scene into 
+   the filesystem dock to create a new scene. This will convert the node
+   in the main scene to an instance of the new scene.
+2. **Duplicate the NPC**: Create a second troll by right-clicking the 
+   ``NPC.tscn`` scene in the filesystem dock and selecting "Duplicate". 
+3. **Rename Nodes**: Name the nodes ``West Troll`` the nodes ``East Troll``. Place the tokens on the left and right sides of the scene, respectively.
 
 
+Troll Personalities
+^^^^^^^^^^^^^^^^^^^
 
-Conversing with the NPC
--------------------------
+1. **Edit East Troll ChatDescr**: Find the ``ChatDescr`` 
+   in the properties panel. Here, input the description: "I engage in a 
+   passionate debate with the West Troll over why my side of the river is 
+   superior."
 
-Include a trigger that starts a conversation when the player interacts with the NPC. Here's a simplified UI demonstration example in C#:
+2. **Edit West Troll ChatDescr**: Similar to the East Troll, enter a description that 
+   states, "I argue with the East Troll about why my side of the river is 
+   far better."
+
+3. **Save and Run**: Save the scene and run your game to ensure that the 
+   troll NPCs are in place and ready to argue.
+
+Modify ``MainGame`` to connect the two trolls together.
 
 .. code-block:: csharp
 
    using Godot;
-   using System;
+   using Microsoft.SemanticKernel;
 
-   public partial class MinimalUserInterface : Control
+   public partial class MainGame : Node
    {
-       ...
-       // Signal handler for when the submit button is pressed
-       public void OnSubmitPressed()
+       public override void _Ready()
        {
-           // Grab the user text
-           string userText = _textInput.Text;
+           GD.Print("Semantic Kernel is ready!");
 
-           // Clear the LineEdit
-           _textInput.Text = "";
+           // Get the troll ChatEntities
+           ChatEntity eastTroll = GetNode<ChatEntity>("East Troll");
+           ChatEntity westTroll = GetNode<ChatEntity>("West Troll");

```

Comment:
If changing the names of the nodes, will also have to change these lines to get nodes `"EastTroll"` and `"WestTroll"`.

Reply:
```suggestion
           ChatEntity eastTroll = GetNode<ChatEntity>("EastTroll");
           ChatEntity westTroll = GetNode<ChatEntity>("WestTroll");
```